### PR TITLE
Use PSR-4 autoloading

### PR DESCRIPTION
**Pull request by vinkla**
1. Switched to PSR-4 autoloading
2. Removed default type attribute
4. Update phpunit.xml.dist config
5. Moved files to root folders

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,10 +1,8 @@
 {
-    "name":     "jsvrcek/ics",
-    "type":     "library",
+    "name": "jsvrcek/ics",
     "description": "abstraction layer for creating multi-byte safe RFC 5545 compliant .ics files",
     "keywords": ["ical", "calendar", "export", ".ics", "RFC 5545", "multi-byte safe"],
     "license": "MIT",
-    "homepage": "https://github.com/jasvrcek/ICS",
```
Comment (jasvrcek) [inline composer.json]: @vinkla thanks for working on this. Before I merge the pull request, is there a reason to remove the link to the github repo?

**Comment by vinkla** [comment]
Yeah, Packagist use GitHub as default homepage URL.

**Comment by jasvrcek** [comment]
Even though it is not necessary in the context of Packagist, I would prefer to keep the homepage url to explicitly indicate where the homepage of the project is, and where people should go to submit issues, etc. Any objections?

**Comment by vinkla** [comment]
Packagist will automagically find the homepage, issues and source URLs by itself.

**Comment by jasvrcek** [comment]
I understand Packagist will find all of that. However, what if a developer
has this library in their project, and looks at the composer.json file to
see where the project home is? For example, say I install
https://github.com/jasvrcek/IcsBundle - this will also install
https://github.com/jasvrcek/ICS as a dependency. Then, imagine at some
point I have an issue with the ICS library - it's reasonable to assume the
composer.json should tell me, as a human reader, where the project home is.

I also notice many of the most popular packages on Packagist do include a
homepage attribute, even if their homepage is github (for example
https://packagist.org/packages/psr/log and
https://packagist.org/packages/monolog/monolog), so keeping it in there is
not going against common practice from what I see.

**Comment by vinkla** [comment]
I've added back the homepage URL.

**Pull request merged**